W-12306411-indexSelectorImprovement-duke (#220)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-selectors.adoc

| Descendants | `..keyName` | Array of values of any matching descendant keys
| Dynamic | See <<dynamic_selector>>. | &nbsp;
| Key-value pair | `.&keyName` | Object with the matching key

> | Index | `[<index>]` | Value of any type at selected array index. Use negative numbers to index from the end of an array, for example, `-1` for the last element in the array.

Author: ```suggestion
| Index | `[<index>]` | Value of any type at selected array index. Use negative numbers to index from the end of an array, for example, `-1` for the last element in the array. Use of numbers beyond the array size returns `null`.
```

> | Index | `[<index>]` | Value of any type at selected array index. To start indexing the last item of the Array, use `-1` because `0` is reserved as the first element in the Array and `-0` does not exist.

Reviewer: I'd simplify this as `Negative numbers will index from the end of the array onwards`. I also think we should say what happens using an index beyond the array size.

Author: revised based on the suggestion above

> | Index | `[<index>]` | Value of any type at selected array index. To start indexing the last item of the Array, use `-1` because `0` is reserved as the first element in the Array and `-0` does not exist.

Author: ```suggestion
| Index | `[<index>]` | Value of any type at selected array index. Use negative numbers to index from the end of an array, starting with `-1` for the last element in the array.
```

> | Index | `[<index>]` | Value of any type at selected array index. `0` is reserved as the first element in the Array and `-0` does not exist, so DataWeave starts indexing the last item of the Array from `-1`.

Author: ```suggestion
| Index | `[<index>]` | Value of any type at selected array index. Note that DataWeave starts indexing the last item of the Array from `-1` because `0` is reserved as the first element in the Array and `-0` does not exist.
```
